server/src/middleware/rpc/layer/rpc_service.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! JSON-RPC service middleware.
+
+use super::ResponseFuture;
+use std::sync::Arc;
+
+use crate::middleware::rpc::RpcServiceT;
+use futures_util::future::BoxFuture;
+use jsonrpsee_core::server::{
+	BoundedSubscriptions, MethodCallback, MethodResponse, MethodSink, Methods, SubscriptionState,
+};
+use jsonrpsee_core::traits::IdProvider;
+use jsonrpsee_types::error::{reject_too_many_subscriptions, ErrorCode};
+use jsonrpsee_types::{ErrorObject, Request};
+
+/// JSON-RPC service middleware.
+#[derive(Clone, Debug)]
+pub struct RpcService {
+	conn_id: usize,
+	methods: Methods,
+	max_response_body_size: usize,
+	cfg: RpcServiceCfg,
+}
+
+/// Configuration of the RpcService.
+#[derive(Clone, Debug)]
+pub(crate) enum RpcServiceCfg {
+	/// The server supports only calls.
+	OnlyCalls,
+	/// The server supports both method calls and subscriptions.
+	CallsAndSubscriptions {
+		bounded_subscriptions: BoundedSubscriptions,
+		sink: MethodSink,
+		id_provider: Arc<dyn IdProvider>,
+		_pending_calls: tokio::sync::mpsc::Sender<()>,
+	},
+}
+
+impl RpcService {
+	/// Create a new service with doesn't support subscriptions.
```

Comment:
Is the comment wrong? Looks like it can support subscriptions if you pass the right `RpcServiceCfg` in?

Reply:
yes